x-pack/plugins/apm/server/lib/services/get_service_transaction_group_comparison_statistics.ts

```diff
@@ -139,44 +142,125 @@ export async function getServiceTransactionGroupComparisonStatistics({
+            y: timeseriesBucket.throughput_rate.value,
+          })
+        );
+        const errorRate = bucket.timeseries.buckets.map((timeseriesBucket) => ({
+          x: timeseriesBucket.key,
+          y: calculateTransactionErrorPercentage(
+            timeseriesBucket[EVENT_OUTCOME]
+          ),
+        }));
+        const transactionGroupTotalDuration =
+          bucket.transaction_group_total_duration.value || 0;
+        return {
+          transactionName,
+          latency,
+          throughput,
+          errorRate,
+          impact: totalDuration
+            ? (transactionGroupTotalDuration * 100) / totalDuration
+            : 0,
+        };
+      });
     }
   );
 }
+
+export async function getServiceTransactionGroupComparisonStatisticsPeriods({
+  serviceName,
+  transactionNames,
+  setup,
+  numBuckets,
+  searchAggregatedTransactions,
+  transactionType,
+  latencyAggregationType,
+  comparisonStart,
+  comparisonEnd,
+  environment,
+}: {
+  serviceName: string;
+  transactionNames: string[];
+  setup: Setup & SetupTimeRange;
+  numBuckets: number;
+  searchAggregatedTransactions: boolean;
+  transactionType: string;
+  latencyAggregationType: LatencyAggregationType;
+  comparisonStart?: number;
+  comparisonEnd?: number;
+  environment?: string;
+}) {
+  const { start, end } = setup;
+
+  const commomProps = {
+    setup,
+    serviceName,
+    transactionNames,
+    searchAggregatedTransactions,
+    transactionType,
+    numBuckets,
+    latencyAggregationType: latencyAggregationType as LatencyAggregationType,
+    environment,
+  };
+
+  const currentPeriodPromise = getServiceTransactionGroupComparisonStatistics({
+    ...commomProps,
+    start,
+    end,
+  });
+
+  const previousPeriodPromise =
+    comparisonStart && comparisonEnd
+      ? getServiceTransactionGroupComparisonStatistics({
+          ...commomProps,
+          start: comparisonStart,
+          end: comparisonEnd,
+        }).then((previousStatistics) => {
+          return previousStatistics.map(
+            ({ transactionName, errorRate, throughput, latency, impact }) => {
+              return {
+                transactionName,
+                impact,
+                errorRate: offsetPreviousPeriodCoordinates({
+                  currentPeriodStart: start,
+                  previousPeriodStart: comparisonStart,
+                  previousPeriodTimeseries: errorRate,
+                }),
+                throughput: offsetPreviousPeriodCoordinates({
+                  currentPeriodStart: start,
+                  previousPeriodStart: comparisonStart,
+                  previousPeriodTimeseries: throughput,
+                }),
+                latency: offsetPreviousPeriodCoordinates({
+                  currentPeriodStart: start,
+                  previousPeriodStart: comparisonStart,
+                  previousPeriodTimeseries: latency,
+                }),
+              };
+            }
+          );
+        })
+      : [];
```

Comment:
it'd be good to have this in a separate function I think, and would this improve if you supplied the offset to `getServiceTransactionGroupComparisonStatistics`?